src/libraries/System.Collections.Immutable/tests/System.Collections.Immutable.Tests.csproj

```diff
@@ -3,13 +3,20 @@
     <NoWarn>$(NoWarn);0436</NoWarn>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkMinimum)</TargetFrameworks>
     <DebuggerSupport Condition="'$(DebuggerSupport)' == '' and ('$(TargetOS)' == 'browser' or '$(TargetOS)' == 'wasi')">true</DebuggerSupport>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>
 
   <ItemGroup>
     <RdXmlFile Include="default.rd.xml" />
   </ItemGroup>
   <ItemGroup>
     <Compile Include="$(CommonTestPath)System\Collections\DictionaryExtensions.cs" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'" Link="Common\System\Collections\DictionaryExtensions.cs" />
+    <!-- Reference internal product components for testing -->
+    <Compile Include="..\src\System\Polyfills.cs" />
+    <Compile Include="..\src\System\Collections\Frozen\String\Hashing.cs" Link="System\Collections\Frozen\String\Hashing.cs" />
+    <Compile Include="..\src\System\Collections\Frozen\String\KeyAnalyzer.cs" Link="System\Collections\Frozen\String\KeyAnalyzer.cs" />
+    <Compile Include="..\src\System\Collections\Immutable\IBinaryTree.cs" Link="System\Collections\Immutable\IBinaryTree.cs" />
+    <Compile Include="..\src\Validation\Requires.cs" Link="Validation\Requires.cs" />
```

Comment:
We should probably follow up to separate these out further, but for now this PR is definitely a good step in the right direction.